### ISSUE TITLE
Incorrect expected indentation for multiline ternary within function invocation

### ISSUE THREAD

**Issue by solmsted** [eslint/eslint#7641]
**Tell us about your environment**

* **ESLint Version:** 3.10.2
* **Node Version:** 6.9.1
* **npm Version:** 4.0.2

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**What did you do? Please include the actual source code causing the issue.**

```js
/*eslint multiline-ternary: [
    "error",
    "always"
]*/
/*eslint indent: [
    "error",
    4,
    {
        CallExpression: {
            arguments: 1
        },
        FunctionDeclaration: {
            body: 1,
            parameters: 1
        },
        FunctionExpression: {
            body: 1,
            parameters: 1
        },
        MemberExpression: 1,
        outerIIFEBody: 1,
        SwitchCase: 1,
        VariableDeclarator: 1
    }
]*/

var f0 = function () {},
    f1 = function (o) {
        return f0(Object.assign(
            {},
            typeof o === 'object' ?
                o :
                {
                    property0: 'abc'
                },
            {
                property1: 'xyz'
            }
        ));
    };

f1({
    property0: '123'
});
```

**What did you expect to happen?**
I did not expect an error for this code.  I expected both of the expressions of the multiline ternary to be indented 4 spaces further then the ternary's condition.

**What actually happened? Please include the actual, raw output from ESLint.**

33:17 - Expected indentation of 12 spaces but found 16. (indent)
34:21 - Expected indentation of 16 spaces but found 20. (indent)
35:17 - Expected indentation of 12 spaces but found 16. (indent)

It thinks the 2nd expression is indented too far.  I also find it strange that there is no error for the first expression.  Should the two expressions not always be expected at the same level of indentation and either both pass or both fail when they both begin at the same column?

**Comment by vitorbal** [comment]
@not-an-aardvark could this be something that would also be handled with your indent rewrite PR?

**Comment by not-an-aardvark** [comment]
Yes, this is fixed in the rewritten version of the rule.

For what it's worth, I think this has the same internal cause as https://github.com/eslint/eslint/issues/7320.
